summingbird-online/src/main/scala/com/twitter/summingbird/online/option/AllOpts.scala

```diff
@@ -98,8 +99,22 @@ trait SummerBuilder extends Serializable {
 
 /**
  * The SummerConstructor option, set this instead of CacheSize, AsyncPoolSize, etc.. to provide how to construct the aggregation for this bolt
+ * @see [[Summers]] for useful [[SummerWithCountersBuilder]]s.
  */
-case class SummerConstructor(get: SummerBuilder)
+case class SummerConstructor(get: SummerWithCountersBuilder)
+
+trait SummerWithCountersBuilder {
+  def create(counter: Name => Counter with Incrementor): SummerBuilder
```

Comment:
I prefer to keep it this way (or maybe change to just `Incrementor`). It should be the same object, not both of them.